incremental-font-transfer/src/patch_group.rs

```diff
@@ -243,6 +227,63 @@ impl<'a> PatchGroup<'a> {
         }
     }
 
+    #[allow(clippy::too_many_arguments)]
+    fn group_patches(
+        candidates: Vec<PatchUri>,
+        ift_compat_id: Option<CompatibilityId>,
+        iftx_compat_id: Option<CompatibilityId>,
+        full_invalidation: &mut Vec<CandidatePatch>,
+        partial_invalidation_ift: &mut Vec<CandidatePatch>,
+        partial_invalidation_iftx: &mut Vec<CandidatePatch>,
+        no_invalidation_ift: &mut BTreeMap<String, NoInvalidationPatch>,
+        no_invalidation_iftx: &mut BTreeMap<String, NoInvalidationPatch>,
```

Comment:
are these only used to aggregate results? If so we can just create them internally and return them, and I would make a little helper struct with a name like `PatchGrouping` that had these five items as members, e.g.

```rust
struct PatchGrouping {
    full_invalidation: Vec<CandidatePatch>,
    partial_invalidation_ift: Vec<CandidatePatch>,
    partial_invalidation_iftx: Vec<CandidatePatch>,
    no_invalidation_ift: BTreeMap<String, NoInvalidationPatch>,
    no_invalidation_iftx: BTreeMap<String, NoInvalidationPatch>,
}

fn group_patches(
    candidates: Vec<PatchUri>,
    ift_compat_id: Option<CompatibilityId>,
    iftx_compat_id: Option<CompatibilityId>,
) -> PatchGrouping { .. }
```

And then if we get this far we could consider making `group_patches` into a constructor on `PatchGrouping`.